File: backend/app/schemas/objective.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel, ConfigDict


class ObjectiveCreate(BaseModel):
    name: str
    description: Optional[str] = None


class ObjectiveUpdate(BaseModel):
    name: Optional[str] = None
    description: Optional[str] = None


class ObjectiveRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    project_id: int
    name: str
    description: Optional[str]
    created_at: datetime

File: frontend/src/types/objective.ts
export interface Objective {
  id: number
  project_id: number
  name: string
  description: string | null
  created_at: string
}

export interface CreateObjectivePayload {
  name: string
  description?: string
}

export interface UpdateObjectivePayload {
  name?: string
  description?: string
}
